### ISSUE TITLE
Allow sorting github pull request comments by date and allow searching in comments

### ISSUE THREAD

**Issue by TheQuantumPhysicist** [microsoft/vscode#149449]
The comments of PRs (in the bottom panel) seem to have some arbitrary order. Probably by their position in files. There has to be an option to make the newest comments appear or top. I can imagine multiple ways of sorting them, including whether resolved, time and position in file. The developer should be able to choose what suits them. In addition to that, it should be possible to search the comments to quickly find things by keyword. Some indexing 
of comments would help.

**Comment by JonasMa** [comment]
I think this feature will help navigating actionable comments a lot. I see that this feature was originally planned for June 2022. Were there any issues with this / is there anything we can help with?

**Comment by hermannloose** [comment]
When I checked last year, order was based on insertion order + grouping by file, so at the moment extensions can guarantee a stable ordering unless comment threads are to be added incrementally later.

I checked some other places in VS Code for sorting options, it doesn't seem very consistent at the moment:

- the Outline view has sorting options in the overflow menu, with the current order indicated by a checkmark
- the Bookmarks view has sorting options in the context menu on right click, but does not indicate which order is in effect

The Bookmarks context menu also mixes actions targeting the selected bookmark with global actions like sorting and clean-up, which does not seem ideal. I would opt for the approach from the Outline view, if possible.

---

> In addition to that, it should be possible to search the comments to quickly find things by keyword. Some indexing of comments would help.

Filtering has been added in #158264 in the meantime.
